src/processor/intents/spawns/_charge-energy.js

```diff
@@ -23,13 +41,13 @@ module.exports = function(spawn, roomObjects, cost, bulk, roomController) {
         cost -= neededEnergy;
         bulk.update(i, {energy: i.energy});
     });
-    
+
     if(cost <= 0) {
         return true;
     }
 
     extensions.sort(utils.comparatorDistance(spawn));
```

Comment:
Is sorting still needed if `energyStructures` is set?

Reply:
Good question. As above, my desire was to minimize the footprint of my changes and group them in an isolated way.

This one is more of a gameplay decision that @artch should take. I suppose if the player has specified the structures to use, the sort makes less sense.

Reply:
I don't think that sorting is needed here. This feature supposed to be an advanced mechanics for experienced players, and they'd like an opportunity to specify discharge order.

Reply:
We can do it this way: if `energyStructures` is undefined, then sort order is by default using distance, otherwise `energyStructures` array order is used.

Reply:
That would be the standard createCreep behavior, right?
Maybe we could extend createCreep itself with the 4th optional parameter (object which may contain energyStructures) instead of introducing new api method?

Reply:
@o4kapuk the idea was to not change the existing createCreep and deprecate the method over a long period of time. Supposing of course that spawnCreep is gobbled up by the community.

I'll make the adjustment to only sort if energyStructures is undefined.

Reply:
I am not super familiar with the code. Does that mean I could always provide all extensions/spawns and just use the array to define the order of discharge?

Or will I have to split extensions among multiple `spawnCreep` calls in the same tick?

Reply:
Yes, you can provide all your extensions/spawns and just use the array order to define the order of discharge.

One additional benefit of doing so is that the system will no longer sort on your behalf, making the call less expensive cpu wise.